src/ui/components/search/locationbiascomponent.js

```diff
@@ -32,7 +34,16 @@ export default class LocationBiasComponent extends Component {
      * Recieve updates from storage based on this index
      * @type {StorageKey}
      */
-    this.moduleId = StorageKeys.LOCATION_BIAS;
+    this.core.storage.registerListener({
+      storageKey: StorageKeys.LOCATION_BIAS,
+      eventType: 'update',
+      callback: data => {
+        const searchIsLoading = data.searchState === SearchStates.SEARCH_LOADING;
+        if (!searchIsLoading && !isEqual(data, this.getState('locationBias'))) {
```

Comment:
Checking that the state is not equal before setting it makes a lot of sense, and I wonder if that could be of use in other components